Criada consulta dos dados de envio do produto

ombrodrigo opened the pull request:
Link documentação: [https://developers.mercadolivre.com.br/pt_br/gerenciamento-de-envios](https://developers.mercadolivre.com.br/pt_br/gerenciamento-de-envios)

ombrodrigo committed:
Criada consulta dos dados de envio do produto

Link documentação: [https://developers.mercadolivre.com.br/pt_br/gerenciamento-de-envios](https://developers.mercadolivre.com.br/pt_br/gerenciamento-de-envios)

dilowagner reviewed (commented):
Legal @ombrodrigo 
Mas ja temos as classes City, State, Country no namespace
`
Dsc\MercadoLivre\Requests\Address
`
Acho que seria interessante reutilizar essas classes para nao ficar repetindo o mapeamento. (Tento seguir esse padrao hehe)

Talvez adicionariamos a Municipality ali tambem nesse namespace relacionado ao Endereco.
Coloquei no Requests por ser uma requisicao publica...

ombrodrigo commented:
@dilowagner 
Pensei em reutilizar as classes City, State, etc. Mais não fiz isso por dois motivos:
- Por exemplo, os  campos `geoInformation`, `country` e `cities` da classe `Dsc\MercadoLivre\Requests\Address\State`, não existem na no payload de shipment. No payload existe apenas o campo `id`e `name`;

- Não quebrar o contrato das classes que utilizam o namespace `Dsc\MercadoLivre\Requests\Address`;

Caso você não veja nenhum problema em alterar as classe do namespace `Dsc\MercadoLivre\Requests\Address\*` já realizo as alterações.

Em relação as requisições publicas:
- Todas as requests publicas devem ficar no namespace `Dsc\MercadoLivre\Requests\*`. ?
- Onde devo colocar as request privadas, para manter o padrão? Pergunto porque no caso do shipment, tenho que possuir um `access_token`.

ombrodrigo committed:
Alterado namespace de Municipality e Neighborhood

Alterado o namespace das classes Municipality, Neighborhood de
`Dsc\MercadoLivre\Resources\Shipment` para `Dsc\MercadoLivre\Requests\Address`

dilowagner reviewed (commented):
Boa @ombrodrigo 

Nao vejo problema de adicionar os atributos nas classes existentes, inclusive em outras chamadas já faço isso... basicamente quando nao temos isso no JSON o atributo fica NULL.

Sobre os namespaces.
O `Requests\*` ficam as chamadas publicas da API
O `Resources\*` ficam as "privadas" que necessitam de Access-Token
Acabei seguindo o padrao de nomenclatura utilizado pelo Mercado Livre na epoca hehe

Vou atualizar o CONTRIBUTING.md com essa info, sinceramente acabei esquecendo de por isso

dilowagner reviewed (commented):
Complementando...
Entendo que ainda tem algumas "falhas de design".
Por exemplo a classe Announcement (que talvez deveria estar no Resources). Mas a ideia na epoca foi deixar mais simples... enfim, talvez repensar numa futura versao hehe.

Inclusive ja estava pensando em refazer com os novos recursos da versao mais atual do PHP7+

ombrodrigo commented:
@dilowagner 
Obrigado pela explicação.

Aproveitando, apliquei as alterações de namespace das classes `Municipality` e `Neighborhood`. 

As duas classes agora estão no namespace `Dsc\MercadoLivre\Requests\Address`.

dilowagner reviewed (approved):
Show man... 
Valeu mais uma vez

dilowagner commented:
Gerada nova [release 1.12.0](https://github.com/discovery-tecnologia/dsc-mercado-livre/releases/tag/1.12.0)
